st2reactor/tests/unit/test_sensor_wrapper.py

```diff
@@ -107,9 +107,31 @@ def test_sensor_init_fails_file_doesnt_exist(self):
         trigger_types = ['trigger1', 'trigger2']
         parent_args = ['--config-file', TESTS_CONFIG_PATH]
 
-        expected_msg = 'Failed to load sensor class from file'
-        self.assertRaisesRegexp(ValueError, expected_msg, SensorWrapper,
+        expected_msg = 'Failed to load sensor class from file.*? No such file or directory'
+        self.assertRaisesRegexp(IOError, expected_msg, SensorWrapper,
                                 pack='core', file_path=file_path,
                                 class_name='TestSensor',
                                 trigger_types=trigger_types,
                                 parent_args=parent_args)
+
+    def test_sensor_init_fails_sensor_code_contains_typo(self):
+        file_path = os.path.join(RESOURCES_DIR, 'test_sensor_with_typo.py')
+        trigger_types = ['trigger1', 'trigger2']
+        parent_args = ['--config-file', TESTS_CONFIG_PATH]
+
+        expected_msg = 'Failed to load sensor class from file.*? \'typobar\' is not defined'
+        self.assertRaisesRegexp(NameError, expected_msg, SensorWrapper,
+                                pack='core', file_path=file_path,
+                                class_name='TestSensor',
+                                trigger_types=trigger_types,
+                                parent_args=parent_args)
+
+        # Verify error message also contains traceback
+        try:
+            SensorWrapper(pack='core', file_path=file_path, class_name='TestSensor',
+                          trigger_types=trigger_types, parent_args=parent_args)
```

Comment:
You want a self.fail('Should have thrown an exception'). Otherwise, tests might still pass on an edge case :P